core/field_checkbox.ts

```diff
@@ -213,12 +216,9 @@ export class FieldCheckbox extends Field<boolean> {
    * @param value The value to convert.
    * @returns The converted value.
    */
-  private convertValueToBool_(value: AnyDuringMigration): boolean {
-    if (typeof value === 'string') {
-      return value === 'TRUE';
-    } else {
-      return !!value;
-    }
+  private convertValueToBool_(value: FieldBool|null): boolean {
+    if (value === null) return false;
+    return parseLegacyBool(value);
```

Comment:
Updated! Yeah, the difference previously was in how `getValueBoolean` and `convertValueToBool_` handled the `null` case since the type reflected possibly being a string, though with the update to `value_`'s type, checking for the string only needs to be done here now anyway.